service/history/shardContext.go

```diff
@@ -294,6 +295,13 @@ func (s *shardContextImpl) AppendHistoryEvents(request *persistence.AppendHistor
 	return err0
 }
 
+func (s *shardContextImpl) NotifyNewHistoryEvent(event *historyEventNotification) error {
```

Comment:
Do we really need this API now? 
Considering it is not doing anything, I would remove it for now.

Reply:
this acts like the abstract function in Java, so it can not be removed.